apsuite/commissioning_scripts/meas_coupling_tune.py

```diff
@@ -0,0 +1,169 @@
+"""Coupling Measurement from Minimal Tune Separation."""
+
+import numpy as _np
+import matplotlib.pyplot as _plt
+
+
+from ..optimization import SimulAnneal as _SimulAnneal
+
+
+class FitTunes(_SimulAnneal):
+    """Tune Fit.
+
+    tunex = coeff1 * quad_parameter + offset1
+    tuney = coeff2 * quad_parameter + offset1
+
+    tune1, tune2 = Eigenvalues([[tunex, coupling/2], [coupling/2, tuney]])
+
+    fit parameters: coeff1, offset1, coeff2, offset2, coupling
+
+    Ex.:
+            fittune = FitTunes(
+                param=quad_strengths, tune1=meas_tune1, tune2=meas_tune2)
+            fittune.fitting_plot()
+            fittune.niter = 1000
+            fittune.start()
+            fittune.fitting_plot()
+    """
+
+    def __init__(self, param, tune1, tune2):
+        """."""
+        self._param = _np.array(param)
+        self._tune1 = _np.array(tune1)
```

Comment:
it is better to use `_np.asarray(tune1)` here. It avoids unnecessary copies in case `tune1` already is an `numpy.ndarray`.

Reply:
I did not know this function. thanks!

Reply:
on the other hand, since I need to sort tune arrays, I think it is safer to work with argument copies by invoking __np.array_. I think the additional memory/process overhead pays for reducing potential side effects of changing class user's data.

Reply:
I changed the code so as not to incur in unnecessary copies, when possible. see what you think...